packages/ab-core/src/types.ts

```diff
@@ -71,7 +71,7 @@ type ListenerFunction = (f: () => void) => void;
 
 export interface Variant {
 	id: string;
-	test: (x: Record<string, unknown>) => void;
+	test: (x: Record<string, unknown>) => void | undefined;
```

Comment:
I think if you want to coerce the return type to not return anything, you would have to only use `undefined`

```suggestion
	test: (x: Record<string, unknown>) => undefined;
```

Reply:
Interestingly that's not the case!
See here: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgIICEAqEDOZkDeAUMqcpHgFzIAUAHtQEoQID2UAJgDx5SgDmAGmQBXEAGsQrAO4gAfAEpkAXjnIAbq2AdkAH1EgOEGKAgcA3EQC+RImxB5kUCGBFQQANS07lpGktVCZHscVgAbCAA6MNZ+GgByAAkIAE94pStLO1YHfGdXd3RWcIg4EBU-BWoAI2KIspU1AicXN3KwKBEIc2RM2xD8OGrsKjQsXHxfYjJyCep-RpaCz29rfpzHIZGwACZqDG2K6bIKMGp8tqKSsrXs3OQtiYBmffHHKZITuaW2r21boA